lib/matrix_sdk/api.ex

```diff
@@ -371,6 +371,15 @@ defmodule MatrixSDK.API do
     |> @http_client.do_request()
   end
 
+  @doc """
+  Redacts a room event with a reason.
+  """
+  def redact_room_event(base_url, token, room_id, event_id, transaction_id, reason) do
```

Comment:
```suggestion
  def redact_room_event(base_url, token, room_id, event_id, transaction_id, opt \\ %{}) do
```

The rest of the codebase follows the pattern above (as `reason` is optional), e.g.:

https://github.com/niklaslong/matrix-elixir-sdk/blob/6891b78243b47036ebf29618473e347bc642d7ac/lib/matrix_sdk/api.ex#L489

I've been using maps as they enforce key uniqueness and the options are always in the body of the request.